libc/string_cmp.c

```diff
@@ -0,0 +1,431 @@
+	TEST_ASSERT_LESS_THAN_INT(0, strncmp(&nonZeroAsciiSet[BUFF_SIZE - 1], &asciiSet[BUFF_SIZE - 1], 1));
+	TEST_ASSERT_EQUAL_INT(0, strncmp(&nonZeroAsciiSet[sizeof(nonZeroAsciiSet) - 1], &nonZeroAsciiSet[sizeof(nonZeroAsciiSet) - 1], BUFF_SIZE - 1));
+	TEST_ASSERT_EQUAL_INT(0, strncmp(asciiSet, nonZeroAsciiSet, 0));
+
+	/* Loop which changes the size of compared space to place where asciSet ends */
+	for (i = 1; i < sizeof(asciiSet); i++) {
+		TEST_ASSERT_LESS_THAN_INT(0, strncmp(asciiSet, nonZeroAsciiSet, i));
+	}
+
+	/* Testing size bigger than string having in mind the compare will end with null terminator */
+	TEST_ASSERT_EQUAL_INT(0, strncmp(asciiSet, asciiSet, SIZE_MAX));
+}
+
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_strcmp)
+{
+}
+
+
+TEST_TEAR_DOWN(string_strcmp)
+{
+}
+
+
+TEST(string_strcmp, basic)
+{
+	TEST_ASSERT_EQUAL_INT(0, strcmp(asciiSet, asciiSet));
+	TEST_ASSERT_LESS_THAN_INT(0, strcmp(asciiSet, nonZeroAsciiSet));
+	TEST_ASSERT_GREATER_THAN_INT(0, strcmp(nonZeroAsciiSet, asciiSet));
+
+	/* We avoid the first place of ucharSet because it is a null terminator which means strcmp will stop its compare */
+	TEST_ASSERT_LESS_THAN_INT(0, strcmp(&ucharEmpty[1], &ucharSet[1]));
+}
+
+
+TEST(string_strcmp, unsignedChar)
+{
+	char holderOutOfAscii[UCHAR_MAX] = { 0 };
+
+	/*
+	 * Loop for creating a buffer similar to ucharSet but with negative values
+	 * to check the ability of strcmp to convert elements to unsigned char
+	 */
+	for (int i = 0; i < sizeof(holderOutOfAscii); i++) {
+		holderOutOfAscii[i] = outOfAsciiSet[sizeof(outOfAsciiSet) - 1 - i];
+	}
+
+	/*
+	 * From lack of defined space size on which CMP is done we are forced to
+	 * set up the last elements of arrays as NULL terms to indicate
+	 * stop point
+	 */
+	ucharSet[sizeof(ucharSet) - 1] = 0;
+	holderOutOfAscii[sizeof(holderOutOfAscii) - 1] = 0;
+
+	TEST_ASSERT_EQUAL_INT(0, strcmp(ucharSet, holderOutOfAscii));
+}
+
+
+TEST(string_strcmp, emptyInput)
+{
+	TEST_ASSERT_EQUAL_INT(0, strcmp(empty, asciiSet));
+	TEST_ASSERT_LESS_THAN_INT(0, strcmp(empty, nonZeroAsciiSet));
+	TEST_ASSERT_EQUAL_INT(0, strcmp(asciiSet, empty));
+	TEST_ASSERT_EQUAL_INT(0, strcmp(empty, empty));
+}
+
+
+TEST(string_strcmp, big)
+{
+	TEST_ASSERT_EQUAL_INT(0, strcmp(hugeString, hugeString));
+	TEST_ASSERT_LESS_THAN_INT(0, strcmp(hugeZeroString, hugeString));
+	TEST_ASSERT_EQUAL_INT(0, strcmp(hugeZeroString, hugeZeroString));
+}
+
+
+TEST(string_strcmp, sizes)
+{
+	TEST_ASSERT_LESS_THAN_INT(0, strcmp(&asciiSet[BUFF_SIZE / 2], &nonZeroAsciiSet[BUFF_SIZE / 2]));
+	TEST_ASSERT_GREATER_THAN_INT(0, strcmp(&asciiSet[sizeof(asciiSet) - 1], &nonZeroAsciiSet[sizeof(nonZeroAsciiSet) - 1]));
+}
+
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_strcoll)
+{
+	errno = 0;
```

Comment:
Why only in tests of strcoll function? Unfortunately it's not certain that errno won't change during the execution of some internal functions:
![Screenshot from 2023-06-27 16-03-55](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/a0e9c058-3fe7-4d49-a044-a913c9c12cdb)